python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -895,6 +926,86 @@ def _pack_value(
     raise SerializationError(f"Unhandled value type {tval}")
 
 
+class _SerdesObjectWrapper(dict):
+    def __init__(self, obj, whitelist_map):
+        self._obj = obj
+        self._whitelist_map = whitelist_map
+        # populate backing native dict to work around c optimizations in json serializer
+        super().__init__({"__serdes": "object_wrapper"})
+
+    def items(self):
+        klass_name = self._obj.__class__.__name__
+        serializer = self._whitelist_map.get_object_serializer(klass_name)
+        yield from serializer.wrap_iter(self._obj, self._whitelist_map)
```

Comment:
tried using a plain class and the json serializer `default` hook - was less performant than this approach